packages/open-next/src/core/routing/cacheInterceptor.ts

```diff
@@ -0,0 +1,202 @@
+  }
+  // calculate age
+  const age = Math.round((Date.now() - (lastModified ?? 0)) / 1000);
+  const hash = (str: string) => createHash("md5").update(str).digest("hex");
+  const etag = hash(body);
+  if (revalidate === 0) {
+    // This one should never happen
+    return {
+      "cache-control":
+        "private, no-cache, no-store, max-age=0, must-revalidate",
+      "x-opennext-cache": "ERROR",
+      etag,
+    };
+  } else if (finalRevalidate !== CACHE_ONE_YEAR) {
+    const sMaxAge = Math.max(finalRevalidate - age, 1);
+    debug("sMaxAge", {
+      finalRevalidate,
+      age,
+      lastModified,
+      revalidate,
+    });
+    const isStale = sMaxAge === 1;
+    if (isStale) {
+      const url = NextConfig.trailingSlash ? `${path}/` : path;
+      await globalThis.queue.send({
+        MessageBody: { host, url },
+        MessageDeduplicationId: hash(`${path}-${lastModified}-${etag}`),
+        MessageGroupId: generateMessageGroupId(path),
+      });
+    }
+    return {
+      "cache-control": `s-maxage=${sMaxAge}, stale-while-revalidate=${CACHE_ONE_MONTH}`,
+      "x-opennext-cache": isStale ? "STALE" : "HIT",
+      etag,
+    };
+  } else {
+    return {
+      "cache-control": `s-maxage=${CACHE_ONE_YEAR}, stale-while-revalidate=${CACHE_ONE_MONTH}`,
+      "x-opennext-cache": "HIT",
+      etag,
+    };
+  }
+}
+
+async function generateResult(
+  event: InternalEvent,
+  localizedPath: string,
+  cachedValue: CacheValue<false>,
+  lastModified?: number,
+): Promise<InternalResult> {
+  debug("Returning result from experimental cache");
+  let body = "";
+  let type = "application/octet-stream";
+  let isDataRequest = false;
+  switch (cachedValue.type) {
+    case "app":
+      isDataRequest = Boolean(event.headers["rsc"]);
+      body = isDataRequest ? cachedValue.rsc : cachedValue.html;
+      type = isDataRequest ? "text/x-component" : "text/html; charset=utf-8";
+      break;
+    case "page":
+      isDataRequest = Boolean(event.query["__nextDataReq"]);
+      body = isDataRequest
+        ? JSON.stringify(cachedValue.json)
+        : cachedValue.html;
+      type = isDataRequest ? "application/json" : "text/html; charset=utf-8";
+      break;
+  }
+  const cacheControl = await computeCacheControl(
+    localizedPath,
+    body,
+    event.headers["host"],
+    cachedValue.revalidate,
+    lastModified,
+  );
+  return {
+    type: "core",
+    statusCode: 200,
+    body: toReadableStream(body, false),
+    isBase64Encoded: false,
+    headers: {
+      ...cacheControl,
+      "content-type": type,
+      ...cachedValue.meta?.headers,
+    },
+  };
+}
+
+export async function cacheInterceptor(
+  event: InternalEvent,
+): Promise<InternalEvent | InternalResult> {
+  if (
+    Boolean(event.headers["next-action"]) ||
+    Boolean(event.headers["x-prerender-revalidate"])
+  )
+    return event;
+  // We localize the path in case i18n is enabled
+  let localizedPath = localizePath(event);
```

Comment:
Same question as above, `basePath` affecting this?

Reply:
Same as above, actually we probably need to do this here only